### PR TITLE
Avoid too many calls to Poco::Logger::get

### PR DESCRIPTION
**Pull request by kssenii**
### Changelog category (leave one):
- Not for changelog (changelog entry is not required)

### CODE REVIEW

**Comment by robot-ch-test-poll** [comment]
This is an automated comment for commit 4fbf52f2de95983b40921d65c68df6a59d6cd570 with description of existing statuses. It's updated for the latest CI running
The full report is available [here](https://s3.amazonaws.com/clickhouse-test-reports/51266/4fbf52f2de95983b40921d65c68df6a59d6cd570/ci_running.html)
The overall status of the commit is 🟡 pending

<table><thead><tr><th>Check name</th><th>Description</th><th>Status</th></tr></thead>
<tbody><tr><td>CI running</td><td>A meta-check that indicates the running CI. Normally, it's in <b>success</b> or <b>pending</b> state. The failed status indicates some problems with the PR</td><td>🟡 pending</td></tr>
<tr><td>Mergeable Check</td><td>Checks if all other necessary checks are successful</td><td>🟢 success</td></tr>
</table>

**Review by KochetovNicolai — APPROVED** [review]

**Review by KochetovNicolai — APPROVED** [review]

**Review by evillique — APPROVED** [review]

File: src/Interpreters/Cache/Metadata.cpp [hunk 1]
```diff
@@ -53,11 +53,17 @@ KeyMetadata::KeyMetadata(
...
     const std::string & key_path_,
     CleanupQueue & cleanup_queue_,
+    [[maybe_unused]] Poco::Logger * log_,
     bool created_base_directory_)
     : key(key_)
     , key_path(key_path_)
     , cleanup_queue(cleanup_queue_)
     , created_base_directory(created_base_directory_)
+#ifdef ABORT_ON_LOGICAL_ERROR
+    , log(&Poco::Logger::get("Key(" + key.toString() + ")"))
```
Comment (alexey-milovidov) [inline src/Interpreters/Cache/Metadata.cpp:63]: Let's remove it entirely. We should not put a high-cardinality value into the logger name.

Comment (alexey-milovidov) [reply src/Interpreters/Cache/Metadata.cpp:63]: Just put the key name into the messages.

Comment (KochetovNicolai) [reply src/Interpreters/Cache/Metadata.cpp:63]: I don't know, it looks ok to me: only debug and sanitizer builds.

Comment (alexey-milovidov) [reply src/Interpreters/Cache/Metadata.cpp:63]: But it's not intended to put a high-cardinality string into the logger name.

Comment (kssenii) [reply src/Interpreters/Cache/Metadata.cpp]: Removed.

**Review by KochetovNicolai — APPROVED** [review]

**Pull request merged**